modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java

```diff
@@ -250,11 +262,11 @@ private void _updateDDMFormField(
 			latestFormFieldMap, formField.getName());
 
 		if (removed) {
+			formField.setReadOnly(true);
+
 			_setDDMFormFieldRemovedLabel(formField);
 		}
 
-		formField.setReadOnly(true);
-
 		// Nested fields
```

Comment:
@carolmariaabb I changed this because at the time of editing the fields that are readonly were going with an empty value. To resolve this, I made only the removed fields readonly.